src/main/scala/com/databricks/spark/sql/perf/results.scala

```diff
@@ -223,19 +223,13 @@ object MLParams {
 }
 
 /**
- * Result information specific to MLlib.
+ * Metrics specific to MLlib benchmark.
  *
- * @param trainingTime  (MLlib) Training time.
- *                      executionTime is set to the same value to match Spark Core tests.
- * @param trainingMetric  (MLlib) Training metric, such as accuracy
- * @param testTime  (MLlib) Test time (for prediction on test set, or on training set if there
- *                  is no test set).
- * @param testMetric  (MLlib) Test metric, such as accuracy
- * @param additionalTests  (MLlib) Additional methods test results.
+ * @param metricName the name of the metric
+ * @param metricValue the value of the metric
+ * @param isLargerBetter  (MLlib) Test metric, such as accuracy
  */
-case class MLResult(
-    trainingTime: Option[Double] = None,
-    trainingMetric: Option[Double] = None,
-    testTime: Option[Double] = None,
-    testMetric: Option[Double] = None,
-    additionalTests: Map[String, Double])
+case class MLMetrics(
+    metricName: String = null,
```

Comment:
We should remove the default values and let user specify them explicitly. We can provide a "Invalid" metric to fill in some boilerplate:

~~~scala
object MLMetric {
  val Invalid = MLMetric("Invalid", 0.0, false)
}
~~~